components/motor/gpio/basic_test.go

```diff
@@ -107,7 +107,18 @@ func TestMotorABPWM(t *testing.T) {
 
 		test.That(t, m.GoFor(ctx, 0, 1, nil), test.ShouldBeError, motor.NewZeroRPMError())
 
+		errChan := make(chan error)
+		go func() {
+			goForErr := m.GoFor(ctx, -50, 100, nil)
+			errChan <- goForErr
+		}()
+		time.Sleep(20 * time.Millisecond)
```

Comment:
presumably this is just to give GoFor time to start "moving"? this has the potential to be flaky. better to use one of the helper methods (e.g. utils.WaitForAssertion or something similar) until the m.IsPowered() returns true.
same comment for other places.

Reply:
I'll remove the sleep and find another way, but the above approach won't test what we're trying to do here. GoFor blocks, but we want to test that a call to Stop from another process causes GoFor to exit early. Not sure how we do this without a coroutine